vmngclient/model/configuration/feature_profile/sdwan/service/appqoe.py

```diff
@@ -0,0 +1,212 @@
+
+    group_name: Default[AppnavControllerGroupName] = Field(
+        default=Default(value=AppnavControllerGroupName.ACG_APPQOE), alias="groupName"
+    )
+    appnav_controllers: List[ForwarderController] = Field(alias="appnavControllers")
+
+
+class ForwarderNodeGroup(BaseModel):
+    model_config = ConfigDict(arbitrary_types_allowed=True, populate_by_name=True)
+
+    name: Union[Global[str], Default[ServiceNodeGroupName]]
+    internal: Default[bool] = Default[bool](value=False)
+    service_node: List[ServiceNodeInformation] = Field(alias="serviceNode")
+
+
+class ForwarderRole(BaseModel):
+    model_config = ConfigDict(arbitrary_types_allowed=True, populate_by_name=True)
+
+    appnav_controller_group: List[ForwarderAppnavControllerGroup] = Field(alias="appnavControllerGroup")
+    service_node_group: List[ForwarderNodeGroup] = Field(alias="serviceNodeGroup")
+    service_context: ServiceContext = Field(alias="serviceContext")
+
+
+# Forwarder and Service
+
+
+class ForwarderAndServiceNodeAddress(str, Enum):
+    ADDRESS_192_168_2_2 = "192.168.2.2"
+
+
+class ServiceNodeInformationDefault(BaseModel):
+    address: Default[ForwarderAndServiceNodeAddress] = Default[ForwarderAndServiceNodeAddress](
+        value=ForwarderAndServiceNodeAddress.ADDRESS_192_168_2_2
+    )
+
+
+class ForwarderAndServiceNodeControllerAddress(str, Enum):
+    ADDRESS_192_168_2_1 = "192.168.2.1"
+
+
+class ForwarderAndServiceNodeController(BaseModel):
+    address: Default[ForwarderAndServiceNodeControllerAddress] = Default[ForwarderAndServiceNodeControllerAddress](
+        value=ForwarderAndServiceNodeControllerAddress.ADDRESS_192_168_2_1
+    )
+
+
+class ForwarderAndServiceNodeAppnavControllerGroup(BaseModel):
+    model_config = ConfigDict(arbitrary_types_allowed=True, populate_by_name=True)
+
+    group_name: Default[AppnavControllerGroupName] = Field(
+        default=Default[AppnavControllerGroupName](value=AppnavControllerGroupName.ACG_APPQOE), alias="groupName"
+    )
+    appnav_controllers: List[ForwarderAndServiceNodeController] = Field(alias="appnavControllers")
+
+
+class ForwarderAndServiceNodeGroup(BaseModel):
+    model_config = ConfigDict(arbitrary_types_allowed=True, populate_by_name=True)
+
+    name: Default[ServiceNodeGroupName] = Default[ServiceNodeGroupName](value=ServiceNodeGroupName.SNG_APPQOE)
+    internal: Default[bool] = Default[bool](value=True)
+    service_node: List[ServiceNodeInformationDefault] = Field(alias="serviceNode")
+
+
+class ForwarderAndServiceNodeRole(BaseModel):
+    model_config = ConfigDict(arbitrary_types_allowed=True, populate_by_name=True)
+
+    appnav_controller_group: List[ForwarderAndServiceNodeAppnavControllerGroup] = Field(alias="appnavControllerGroup")
+    service_node_group: List[ForwarderAndServiceNodeGroup] = Field(alias="serviceNodeGroup")
+
+    service_context: ServiceContext = Field(alias="serviceContext")
+
+
+# Service
+class ServiceNodeExternalAddress(str, Enum):
```

Comment:
This looks over complicated.
Why enum classes like `ServiceNodeExternalAddress`, `ServiceNodeExternalVpgIp` are needed? Looks like they store single default value.

Reply:
Models will be rewritten. Merging before `catalystwan` name change.